### ISSUE TITLE
[WISH] e-commerce categories should be explicit in product URL.

### ISSUE THREAD

**Issue by zeroheure** [odoo/odoo#60141]
This is related to ecommerce SEO. Search engines take care of URLs content when indexing ; words founds in URLs become part of the search engine's index. This, if we have explicit categories in URLs instead of numbers it will help SEO.

Impacted versions: 8 to 14

Steps to reproduce: install e-commerce, enable display of ecommerce categories in shop, browse in a category, then choose a product in that category.

Current behavior: the product's url contains the category ID

Expected behavior: product's url with the category slug and ID

**Comment by pedrobaeza** [comment]
Starting with, a product can be in multiple website categories. And AFAIK, that URL analysis is not something up to date with current crawlers.

**Comment by zeroheure** [comment]
Multiple website category is irelevant, because the category ID is only (and allways) added in product URL if you browse from that category. 
It is true that URLs are only a **minor ranking factor**. But adding category name will not increase the load so much. And category name will also help with URL sharing. For URL analysis, see this [DMOZ advice about keywords in URLs](https://moz.com/learn/seo/url)

**Comment by Yenthe666** [comment]
@JKE-be is probably to decide / give good feedback here :)

**Comment by JKE-be** [comment]
need to analyze.

Because one product can have several category, and so you will have several url with only one canonical to the default without category ? 

ps: in case you only have on category, you can change in v14 the slug name to add yourself the category into the slug.

**Comment by zeroheure** [comment]
Canonical URL is allways the one without category ?

**Comment by JKE-be** [comment]
I think... no?

product 'apple' in category 'fruits' and 'green' -> urls are

/shop/green/apple-1
/shop/fruits/apple-1
/shop/apple-1

All with a duplicated content... So you need to have a canonical

the /shop/apple-1 is the only one true in the 3 cases.

I'm wrong ?

**Comment by zeroheure** [comment]
No, you're right. I was wondering if canonical URL could be attached to a product in a category, like in your sample the first or second url. It would give more "heavy" keywords to search engines.
And it might help search engines to choose the website categories to display in SERP infobox (I really don't know, I miss time to check in Google doc).

**Comment by zeroheure** [comment]
The french SEO expert [Olivier Andrieu has done a short video about this problem](https://www.abondance.com/20211019-46447-longueur-durl-et-seo-video-seo-abondance-n207.html). There is a transcript too.

The key point is while categories in URLs have a minor impact in SEO, they are useful for the work on your analytics datas and logs, because you can use the category slug to segment the logs and the analytics.
